src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -270,6 +281,24 @@ async def test_stream_unary_async_generator(self):
         self.assertEqual(_RESPONSE, response)
         self.assertEqual(await call.code(), grpc.StatusCode.OK)
 
+    async def test_stream_unary_async_generator_with_request_iter(self):
+        stream_unary_call = self._channel.stream_unary(_STREAM_UNARY_ASYNC_GEN)
+
+        finished = False
+
+        def request_gen():
+            for _ in range(_NUM_STREAM_REQUESTS):
+                yield _REQUEST
+            nonlocal finished
```

Comment:
are we good with py3-only nonlocal keyword?  I'd been used to doing using a mutable object like a list in py2...

Reply:
Yeah, both syntax looks good to me. Depends on you.

One alternative is use the `asyncio.Event` to tell whether the request generator reaches the end or not: https://docs.python.org/3/library/asyncio-sync.html#event

The `asyncio.Event.set` can be called without `async` keyword.